x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/local/LocalExporter.java

```diff
@@ -138,11 +139,12 @@ boolean isExporterReady() {
     }
 
     @Override
-    public LocalBulk openBulk() {
+    public void openBulk(final ActionListener<ExportBulk> listener) {
         if (state.get() != State.RUNNING) {
-            return null;
+            listener.onResponse(null);
+        } else {
+            listener.onResponse(resolveBulk(clusterService.state(), false));
```

Comment:
The `resolveBulk(...)` method also needs to be made async, because the `setupIfElectedMaster(...)` method makes remote calls in a sync manner. The `setupIfElectedMaster(...)` method collects async remote calls in the `asyncActions` list and then later executes these calls in a sync manner.

Reply:
Most async actions are kind of async already, the `setupIfElectedMaster(...)` method just executes these actions in a sync manner. So I think changing this is less invasive as in http exporter.

Reply:
Mind if I do this as a follow-up PR?

Reply:
Sure, that makes sense.

Reply:
Created #35797 for this.